### PR TITLE
Add `databricks_git_credential` resource

### PR DESCRIPTION
**Pull request by alexott**

### CODE REVIEW

File: exporter/importables.go [hunk 1]
```diff
@@ -818,4 +818,33 @@ var resourcesMap map[string]importable = map[string]importable{
 			return nil
 		},
 	},
+	"databricks_git_credential": {
+		Service: "repos",
+		Name: func(d *schema.ResourceData) string {
+			name := d.Get("git_provider").(string) + "_" + d.Get("user_name").(string) + "_" + d.Id()
+			re := regexp.MustCompile(`[^0-9A-Za-z_]`)
+			return re.ReplaceAllString(name, "_")
```
Comment (nfx) [inline exporter/importables.go:826]: aren't non-alphanum characters filtered already by the framework?

File: repos/resource_git_credential.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+			if err != nil {
+				return err
+			}
+			d.Set("git_provider", resp.Provider)
+			d.Set("user_name", resp.UserName)
+			return nil
+		},
+		Update: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			api := NewGitCredentialAPI(ctx, c)
+			req := GitCredentialRequest{Provider: d.Get("git_provider").(string), UserName: d.Get("user_name").(string),
```
Comment (nfx) [inline repos/resource_git_credential.go:135]: use reflect resource

**Review by nfx — CHANGES_REQUESTED** [review]
+ acceptance test
+ doc page

File: repos/resource_git_credential.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+		Schema:        s,
+		SchemaVersion: 1,
+		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			api := NewGitCredentialAPI(ctx, c)
+			req := GitCredentialRequest{Provider: d.Get("git_provider").(string), UserName: d.Get("user_name").(string),
+				PAT: d.Get("access_token").(string)}
+			resp, err := api.Create(req)
+
+			if err != nil {
+				if !d.Get("force").(bool) || !strings.HasPrefix(err.Error(), "Only one Git credential is supported at this time") {
```
[the remark this answers is not in the document]
Comment (alexott) [reply repos/resource_git_credential.go:103]: it's generic 400, bad request :-(

File: repos/resource_git_credential.go [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+				}
+				creds, err := api.List()
+				if err != nil {
+					return err
+				}
+				if len(creds) != 1 {
+					return fmt.Errorf("list of credentials is either empty or have more than one entry (%d)", len(creds))
+				}
+				resp = creds[0]
+				log.Printf("[INFO] new resp=%v", resp)
```
Comment (nfx) [inline repos/resource_git_credential.go:101]: add a context, e.g. `updating existing git credentials` or something. or remove it.

**Review by nfx — APPROVED** [review]
lgtm :) 

though, need a doc and acceptance test

**Review by nfx — CHANGES_REQUESTED** [review]
see comments

File: repos/resource_git_credential.go [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+	client  *common.DatabricksClient
+	context context.Context
+}
+
+// GitCredentialsAPI creates GitCredentialsAPI instance from provider meta
+func NewGitCredentialsAPI(ctx context.Context, m interface{}) GitCredentialsAPI {
+	return GitCredentialsAPI{m.(*common.DatabricksClient), ctx}
+}
+
+type GitCredentialRequest struct {
```
[the remark this answers is not in the document]
Comment (alexott) [reply repos/resource_git_credential.go:24]: shorter names could be easier to understand, plus, making it more unified with our naming conventions

**Comment by alexott** [comment]
> * acceptance test
> 
>     * doc page

doc will be after ok from PM

**Review by nfx — APPROVED** [review]

**Pull request merged**